edit/lint.js

```diff
@@ -0,0 +1,319 @@
+/* global CodeMirror CSSLint editors makeSectionVisible showHelp showCodeMirrorPopup */
+/* global stylelintDefaultConfig onDOMscripted injectCSS require */
+'use strict';
+
+function initLint() {
+  document.getElementById('lint-help').addEventListener('click', showLintHelp);
+  document.getElementById('lint').addEventListener('click', gotoLintIssue);
+  window.addEventListener('resize', resizeLintReport);
+  document.getElementById('stylelint-settings').addEventListener('click', openStylelintSettings);
+
+  // touch devices don't have onHover events so the element we'll be toggled via clicking (touching)
+  if ('ontouchstart' in document.body) {
+    document.querySelector('#lint h2').addEventListener('click', toggleLintReport);
+  }
+  BG.chromeLocal.getValue('editorStylelintRules').then(rules => setStylelintRules(rules));
+}
+
+function setStylelintRules(rules = '') {
+  if (Object.keys(rules).length === 0 && typeof stylelintDefaultConfig !== 'undefined') {
+    rules = deepCopy(stylelintDefaultConfig.rules);
+  }
+  BG.chromeLocal.setValue('editorStylelintRules', rules);
+}
+
+function getLinterConfigForCodeMirror(name) {
+  return CodeMirror.lint && CodeMirror.lint[name] ? {
+    getAnnotations: CodeMirror.lint[name],
+    delay: prefs.get('editor.lintDelay')
+  } : false;
+}
+
+function updateLinter(name) {
+  function updateEditors() {
+    const options = getLinterConfigForCodeMirror(name);
+    CodeMirror.defaults.lint = options === 'null' ? false : options;
+    editors.forEach(cm => {
+      // set lint to "null" to disable
+      cm.setOption('lint', options);
+      cm.refresh(); // enabling/disabling linting changes the gutter width
```

Comment:
There is only a [`no-inline-comments` rule](https://eslint.org/docs/rules/no-inline-comments).  I'll go ahead and enable it and fix any issues that pop up.